### PR TITLE
Various fixes to bf-lu

### PR DESCRIPTION
**Pull request by vishwacsena**
Fixes address and enable end to end generated dialogs experience.

### CODE REVIEW

File: packages/lu/src/parser/lufile/simpleIntentSection.js [hunk 1]
```diff
@@ -50,13 +50,24 @@ class SimpleIntentSection {
...
-                })))
+                let utteranceAndEntities;
+                try {
+                    utteranceAndEntities = visitor.visitNormalIntentStringContext(normalIntentStr);
+                }
+                catch (err) {
+                    errors.push(BuildDiagnostic({
+                        message: "Invalid utterance definition found. Did you miss a '{' or '}'?",
+                        context: normalIntentStr
+                    }))
```
Comment (feich-ms) [inline packages/lu/src/parser/lufile/simpleIntentSection.js:61]: Only this error will be thrown out? No other error messages?

File: packages/lu/src/parser/lufile/parseFileContents.js [hunk 2]
```diff
@@ -876,37 +879,43 @@ const parseAndHandleSimpleIntentSection = function (parsedContent, luResource) {
...
                                         addItemOrRoleIfNotPresent(parsedContent.LUISJsonStructure, LUISObjNameEnum.CLOSEDLISTS, entity.entity, [entity.role.trim()]);
                                     } else {
-                                        let errorMsg = `${entity.entity} has been defined as a LIST entity type. It cannot be explicitly included in a labelled utterance unless the label includes a role.`;
-                                        let error = BuildDiagnostic({
-                                            message: errorMsg,
-                                            context: utteranceAndEntities.context
-                                        });
-
-                                        throw (new exception(retCode.errorCode.INVALID_INPUT, error.toString(), [error]));
+                                        if (!isChildEntity(entity, entitiesFound)) {
```
Comment (munozemilio) [inline packages/lu/src/parser/lufile/parseFileContents.js:882]: This operation is repetitive and can be a function that receives the error message

**Comment by munozemilio** [comment]
@vishwac can you please provide what the fixes are about and provide some release notes as well including issues

**Comment by vishwacsena** [comment]
@munozemilio sure
- [x] BF-LU parser now throws an error if there are mismatched number of braces in an utterance. 
```
 add {{@AddToQuantity={@number=99}}, {{@AddToQuantity={@number=215}} and {{@AddToQuantity={@number=99}}
```
- [x] Various fixes related to merging as well as validations for new nDepth (hierarchical ML) entity
    - [x] Upon merging (collate), nDepth entities with incompatible types throw
    - [x] Upon merging (collate), nDepth entities are correctly merged via reference
    - [x] Added logic to verify ml entity both at root as well as child level is handled correctly
    - [x] Added validation to ensure every child must have its parent labelled in an utterance
    - [x] Relaxed validations to allow prebuilt child entity type to be labelled in an utterance
    - [x] Relaxed validations to allow list child entity type to be labelled in an utterance

**Pull request merged**